src/fplan/fplan.py

```diff
@@ -193,7 +197,8 @@ def solve(args):
 
         (taxbase, last_cut, last_rate) = (0, 0, 0)
         for (cut, rate) in S.taxrates:
-            rate += S.state_tax
+            if rate > 0:                             # if below fed std_ded, assumes tax 0%
```

Comment:
This tweak needs to be copied to the print function below.

Reply:
Assuming that the federal taxes don't add a 0% bracket at a non-zero dollar amount, the practical effect of this change is to add a [0, 0] bracket to state taxes, just like we do with federal taxes.  It ensures that we don't get a state refund for staying under the standard deduction.

The print function already makes an equivalent assumption in lines 408-409 where it sets the inc (and therefore taxes) to 0 if it is below the standard deduction.

The problem with this code is that it assumes that the state standard deduction is the same as the federal standard deduction.  That is not universally true.  Maybe we need another configuration parameter (I think there was a state deduction variable, but it wasn't used anywhere).  We could try to make the code clearer if we add that.

Reply:
Ok, ignore most of my last comment.  Your fix was needed.